src/lib/operators/join_hash.cpp

```diff
@@ -469,30 +449,75 @@ class JoinHash::JoinHashImpl : public AbstractJoinOperatorImpl {
           Hence we are going to write NULL values for each row.
           */
 
-          auto pos_list_left_local = std::make_shared<PosList>();
-          auto pos_list_right_local = std::make_shared<PosList>();
-
           for (size_t partition_offset = partition_begin; partition_offset < partition_end; ++partition_offset) {
             auto& row = partition[partition_offset];
-            pos_list_left_local->emplace_back(RowID{ChunkID{0}, INVALID_CHUNK_OFFSET});
-            pos_list_right_local->emplace_back(row.row_id);
+            pos_list_left_local.emplace_back(RowID{ChunkID{0}, INVALID_CHUNK_OFFSET});
+            pos_list_right_local.emplace_back(row.row_id);
           }
-          if (!pos_list_left_local->empty()) {
-            pos_list_left[current_partition_id] = pos_list_left_local;
-            pos_list_right[current_partition_id] = pos_list_right_local;
+        }
+
+        if (!pos_list_left_local.empty()) {
+          pos_list_left[current_partition_id] = std::move(pos_list_left_local);
+          pos_list_right[current_partition_id] = std::move(pos_list_right_local);
+        }
+      }));
+      jobs.back()->schedule();
+    }
+
+    CurrentScheduler::wait_for_tasks(jobs);
+  }
+
+  void _probe_semi_anti(const RadixContainer<RightType>& radix_container,
+                        const std::vector<std::shared_ptr<HashTable<LeftType>>>& hashtables,
+                        std::vector<PosList>& pos_lists) {
+    std::vector<std::shared_ptr<AbstractTask>> jobs;
+    jobs.reserve(radix_container.partition_offsets.size() - 1);
+
+    for (size_t current_partition_id = 0; current_partition_id < (radix_container.partition_offsets.size() - 1);
+         ++current_partition_id) {
+      jobs.emplace_back(std::make_shared<JobTask>([&]() {
+        // Get information from work queue
+        auto& partition = static_cast<Partition<RightType>&>(*radix_container.elements);
+        const auto& partition_begin = radix_container.partition_offsets[current_partition_id];
+        const auto& partition_end = radix_container.partition_offsets[current_partition_id + 1];
+
+        // Skip empty partitions to avoid empty output chunks
+        if ((partition_end - partition_begin) == 0) {
+          return;
+        }
+
+        // auto pos_list_local = std::make_shared<PosList>();
```

Comment:
Is this supposed to be here?

Reply:
is anyone really where they're supposed to be? think about it

Reply:
done